docs/rfcs/0004-agent-flow.md

```diff
@@ -0,0 +1,54 @@
+# Agent Flow - Agent Utilizing Components 
+
+* Date: 2022-03-30
+* Author: Matt Durham (@mattdurham)
+* PR: [grafana/agent#1538](https://github.com/grafana/agent/pull/1538)
+* Status: Draft
+
+## Overarching Problem Statement
+
+The Agents configuration and onboarding is difficult to use. Viewing the effect of configuration changes on telemetry data is difficult. Making the configuration simplier, composable and intuitive allieviates these concerns.
```

Comment:
I specifically called out *onboarding* here, meaning the new user experience. I think whether we want to include this word can have significant changes to the rest of the design doc.